greedy_methods/fractional_knapsack.py

```diff
@@ -1,27 +1,54 @@
 from bisect import bisect
 from itertools import accumulate
 
+# Reference Link: https://en.wikipedia.org/wiki/Continuous_knapsack_problem
 
-def frac_knapsack(vl, wt, w, n):
+
+def frac_knapsack(
+    values: list[int], weights: list[int], capacity: int, max_items: int
+) -> float:
     """
+    This function implements fractional knapsack problem.
+
+    Args:
+        values: List of values of items.
+        weights: List of weights of items.
+        capacity: Capacity of the knapsack.
+        max_items: Number of items.
+
+    Returns:
+        Maximum value of items that can be put into the knapsack.
+
     >>> frac_knapsack([60, 100, 120], [10, 20, 30], 50, 3)
     240.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 10, 4)
     105.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 4)
     95.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6], 8, 4)
```

Comment:
This looking much better but is the _fractional_ part working?

https://www.geeksforgeeks.org/fractional-knapsack-problem says that
```
>>> frac_knapsack([500], [3], 10, 1)
166.66667
```
but we get `500`

Reply:
You are right... the existing algo does not implement fractional knapsack correctly.